sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetFileFormat.scala

```diff
@@ -105,29 +105,60 @@ object GpuParquetFileFormat {
     }
   }
 
-  def parquetWriterOptionsFromSchema[T <: NestedBuilder[_, _], V <: ParquetColumnWriterOptions]
-  (builder: ParquetColumnWriterOptions.NestedBuilder[T, V],
-   schema: StructType, writeInt96: Boolean): T = {
-    // TODO once https://github.com/rapidsai/cudf/issues/7654 is fixed go back to actually
-    // setting if the output is nullable or not everywhere we have hard-coded nullable=true
-    schema.foreach(field =>
-      field.dataType match {
+  def parquetWriterOptionsFromField[T <: NestedBuilder[_, _], V <: ParquetColumnWriterOptions](
+      builder: ParquetColumnWriterOptions.NestedBuilder[T, V],
+      dataType: DataType,
+      name: String,
+      writeInt96: Boolean,
+      nullable: Boolean): T = {
+      dataType match {
         case dt: DecimalType =>
-          builder.withDecimalColumn(field.name, dt.precision, true)
+          builder.withDecimalColumn(name, dt.precision, nullable)
         case TimestampType =>
-          builder.withTimestampColumn(field.name,
-            writeInt96, true)
+          builder.withTimestampColumn(name, writeInt96, nullable)
         case s: StructType =>
           builder.withStructColumn(
-            parquetWriterOptionsFromSchema(structBuilder(field.name), s, writeInt96).build())
+            parquetWriterOptionsFromSchema(
+              structBuilder(name, nullable),
+              s,
+              writeInt96).build())
         case a: ArrayType =>
           builder.withListColumn(
-            parquetWriterOptionsFromSchema(listBuilder(field.name),
-              StructType(Array(StructField(field.name, a.elementType, true))), writeInt96)
-              .build())
+            parquetWriterOptionsFromField(
+              listBuilder(name, nullable),
+              a.elementType,
+              name,
+              writeInt96,
+              true).build())
```

Comment:
Why is this true when the others are nullable? A comment would be good.

Reply:
let me know if the comment for passing `nullable` param to the listBuilder and others is enough. We already documented why we are hard-coding nullable to true